UnityGsdk/Assets/PlayFabSdk/MultiplayerAgent/Model/SessionConfig.cs

```diff
@@ -38,6 +41,47 @@ public void CopyNonNullFields(SessionConfig other)
             {
                 InitialPlayers = other.InitialPlayers;
             }
+            
+           if (other.Metadata != null && other.Metadata.Any())
+           {
+               Metadata = other.Metadata;
+           }
+            
+        }
+        
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as SessionConfig);
+        }
+
+        public bool Equals(SessionConfig other)
+        {
+            return other != null &&
+                   SessionId == other.SessionId &&
+                   SessionCookie == other.SessionCookie &&
+                   EqualityComparer<List<string>>.Default.Equals(InitialPlayers, other.InitialPlayers) &&
+                   EqualityComparer<Dictionary<string, string>>.Default.Equals(Metadata, other.Metadata);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = -481859842;
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(SessionId);
```

Comment:
These were generated with Resharper which is considered to have correct method building to generate correct hash codes for equality comparison. These objects are effectively structs but in the form of a class because y'know, JSON and thus their contents define them, not a hash of their instance. This also helps ensures two objects with the same internal data are considered the same so you can't mess up comparing them.

Reply:
gotcha, I'd leave a comment that these are Resharper generated, for the curious reader :)